### DISCUSSION TITLE
Authorization

### DISCUSSION THREAD

**Discussion by zinoubm** [gaganpreet/fastapi-starter#528] [General]
I'm having a hard time with authorization in the swagger, There are 4 fields. password, username, id and client secret which is confusing me, what should I do to authorize myself.

**Comment by gaganpreet** [comment]
I'm not sure which endpoint you're referring to here, can you be more specific here?
FastAPI users also handles OAuth and perhaps you are looking at the `/jwt/` endpoint? The auth endpoints are generated by FastAPI Users, and it might be helpful for you to read their docs as well.

**Reply by zinoubm** [reply]
After registering an account and login, I get a token. Then when I try to create an item from the "/api/v1/items" (POST), I don't see any place where I can provide my token, and If I try to create an item I get "{
  "detail": "Unauthorized"
}", Hope you can help me with that.

**Reply by johnjameswhitman** [reply]
I was having the same issue when trying to authenticate in the Swagger UI. I fixed it by changing the `tokenUrl` from `auth/jwt/login` to `/api/v1/auth/jwt/login` after noticing the Swagger UI was treating it as a relative URL.

https://github.com/gaganpreet/fastapi-starter/blob/0718cea5df21712063b7083906ab1a77ada7159d/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/deps/users.py#L18

**Before**
![Screenshot_20230815_212715](https://github.com/gaganpreet/fastapi-starter/assets/1639083/cd88e42c-699d-4514-9fce-067a786dd5df)

**After**
![Screenshot_20230815_212540](https://github.com/gaganpreet/fastapi-starter/assets/1639083/38e3fa7f-6521-414b-a98e-718189a1d02c)

**Reply by zinoubm** [reply]
@johnjameswhitman  thanks, this will help a lot of people ⭐
